yax/state/exe.py

```diff
@@ -87,7 +87,30 @@ def __init__(self):
         self.details = {'run_key': Str}
 
     @property
-    def artifact_dependencies(self):
+    def bound_artifact_to_upstream_nodes(self):
+        r"""Map bound artifact names to the nodes which spawned them.
+
+        If the pipeline looked like this:
+
+              / -- Node3 ---- Node4
+        Node1 ---- Node2 -- /
+
+        with bound artifacts:
+            Node1: foo
+            Node2: bar
+            Node3: baz
+            Node4: fiz
+
+        The results would be:
+
+        {
+            'foo': {Node1},
+            'bar': {Node2, Node1},
+            'baz': {Node3, Node1},
+            'fiz': {Node4, Node3, Node2, Node1}
+        }
+
+        """
         node_dependencies = {}
         adjm = self.adjacency_matrix
```

Comment:
knit picky but adj_m seems to be more in line with the conventions you are switching to.